src/pykeen/models/base.py

```diff
@@ -263,6 +263,20 @@ def load_state(self, path: Union[str, os.PathLike]) -> None:
 
     """Prediction methods"""
 
+    def _prepare_batch(self, batch: torch.LongTensor, index_relation: int) -> torch.LongTensor:
```

Comment:
this doesn't seem to replace any previous code that handles the inverse triples... will these get out of sync now?

Reply:
In our context, inverse triples are not part of the validation/test triples. However, we can perform inference based on inverse triples by calling `score_hrt_inverse()`, `score_h_inverse()`, and `score_t_inverse()` where inverse triples are generated onn the fly.

Reply:
610a3af

Reply:
For `score_*_inverse`, it seems like we also need to provide the triples where the original ID has been replaced by `2 * r`. I generally think, this should be fine, since the `score_*` methods are more fundamental and give access to the "raw" model, whereas `predict_*` adds some additional logic around it, i.e. inverse triples handling, sigmoid.

Reply:
Besides the `2*r` assumption, the `score_*_inverse` methods will also give unexpected results when called with already inverted relations, since they always add 1 to it, and thus end up with the next normal relation (using the current default inverse relation ID logic).